daemon/api.go

```diff
@@ -586,78 +612,151 @@ func deleteOp(c *Command, r *http.Request) Response {
 }
 
 type packageInstruction struct {
+	progress.NullProgress
 	Action   string `json:"action"`
 	LeaveOld bool   `json:"leave_old"`
 	pkg      string
-	prog     progress.Meter
+	chin     chan interface{}
+	chout    chan interface{}
 }
 
-func (inst *packageInstruction) install() interface{} {
-	flags := snappy.DoInstallGC
-	if inst.LeaveOld {
-		flags = 0
+func (inst *packageInstruction) Agreed(intro, licenseFile string) bool {
+	inst.chout <- map[string]interface{}{
+		"license_ack_needed": true,
+		"license_intro":      intro,
+		"license_text":       licenseFile,
 	}
-	_, err := snappy.Install(inst.pkg, flags, inst.prog)
 
+	b, ok := (<-inst.chin).(bool)
```

Comment:
So, client calls install, snapd downloads, notices it needs a license agreement, so the async install fails with an error message that includes the license text and header, and the client asks the user for license confirmation, and then resubmits the install request with e.g. a x-license-accepted header?